ambry-api/src/test/java/com.github.ambry/rest/RequestPathTest.java

```diff
@@ -61,24 +63,36 @@ public void testRequestPathParsing() {
           String path = prefixToUse + clusterNameToUse + opOrId;
           // the uri as is (e.g. "/expectedOp).
           parseRequestPathAndVerify(path, prefixesToRemove, clusterName,
-              new RequestPath(expectedPrefix, expectedClusterName, opOrId, opOrId, null));
+              new RequestPath(expectedPrefix, expectedClusterName, opOrId, opOrId, null,
+                  NO_BLOB_SEGMENT_IDX_SPECIFIED));
 
           // the uri with a query string (e.g. "/expectedOp?param=value").
           parseRequestPathAndVerify(path + queryString, prefixesToRemove, clusterName,
-              new RequestPath(expectedPrefix, expectedClusterName, opOrId, opOrId, null));
+              new RequestPath(expectedPrefix, expectedClusterName, opOrId, opOrId, null,
+                  NO_BLOB_SEGMENT_IDX_SPECIFIED));
           if (opOrId.length() > 1) {
             for (RestUtils.SubResource subResource : RestUtils.SubResource.values()) {
               String subResourceStr = "/" + subResource.name();
+              //if subResource is "Segment", generate a random integer segment to add to the end of the uri
+              int segment = -1;
+              if (subResource.name().equals("Segment")) {
+                segment = new Random().nextInt(Integer.MAX_VALUE);
```

Comment:
Looks like here we use `int` for segment index. I think @lightningrob has similar comment, do we need `long` type for segment index?

Reply:
changing to int